scene/gui/tab_bar.cpp

```diff
@@ -478,16 +544,22 @@ int TabBar::get_tab_count() const {
 }
 
 void TabBar::set_current_tab(int p_current) {
-	if (current == p_current) {
-		return;
-	}
 	ERR_FAIL_INDEX(p_current, get_tab_count());
 
-	previous = current;
 	current = p_current;
+	emit_signal(SNAME("tab_selected"), current);
+
+	if (current == previous) {
+		return;
+	}
+	previous = current == -1 ? p_current : current;
```

Comment:
`p_current` and `current` are equal at this point, no? What's the point of this condition?